common/SmokeTests/SmokeTest/Update-Dependencies.ps1

```diff
@@ -31,63 +35,79 @@ function Log-Warning($message) {
+    if (!$versions) {
+        Write-Warning "Did not find any versions for $($packageName)"
+        return
+    }
 
-# For each PackageReferecne in the csproj, find the latest version of that
-# package from the dev feed which is not in the excluded list.
-$projectFilePath = Resolve-Path -Path $ProjectFile
-[xml]$csproj = Get-Content $ProjectFile
-$csproj |
-    Select-XML $PACKAGE_REFERENCE_XPATH |
-    Where-Object { $_.Node.HasAttribute('Version') } |
-    ForEach-Object {
-        # Resolve package version:
-        $packageName = $_.Node.Include
-
-        # This assumes that the versions coming back from Find-Package are
-        # sorted ascending. It excludes any version that is in the corresponding
-        # $PACKAGE_EXCLUSIONS entry
-        $targetVersion = ($allPackages |
-            Where-Object { $_.Name -eq $packageName } |
-            Where-Object { -not ( $PACKAGE_EXCLUSIONS.ContainsKey($packageName) -and $PACKAGE_EXCLUSIONS[$packageName].ContainsKey($_.Version)) } |
-            Select-Object -Last 1).Version
-
-        if ($targetVersion -eq $null) {
-            return
-        }
+    $sorted = [AzureEngSemanticVersion]::SortVersionStrings($versions)
+    return $sorted | Select-Object -First 1
+}
 
-        Write-Host "Setting $packageName to $targetVersion"
-        $_.Node.Version = "$targetVersion"
+function SetLatestPackageVersions([object]$csproj) {
+    # For each PackageReference in the csproj, find the latest version of that
+    # package from the dev feed which is not in the excluded list.
+    $allPackages = GetAllPackages
+    $csproj |
+        Select-XML $PACKAGE_REFERENCE_XPATH |
+        Where-Object { $_.Node.HasAttribute('Version') } |
+        ForEach-Object {
+            # Resolve package version:
+            $packageName = $_.Node.Include
+
+            $targetVersion = GetLatestPackage $allPackages $packageName
+
+            if ($targetVersion -eq $null) {
+                return
+            }
 
+            Write-Host "Setting $packageName to $targetVersion"
+            $_.Node.Version = "$targetVersion"
 
-        # Validate package version date component matches
-        if ($SkipVersionValidation) {
-            return
-        }
 
-        if ($_.Node.Version -match $DEV_DATE_REGEX) {
-            if ($baselineVersionDate -eq $null) {
-                Write-Host "Using baseline version date: $($matches[1])"
-                $baselineVersionDate = $matches[1]
+            # Validate package version date component matches
+            if ($SkipVersionValidation) {
+                return
             }
 
-            if ($baselineVersionDate -ne $matches[1]) {
-                Log-Warning "$($_.Node.Include) uses invalid version. Expected: $baselineVersionDate, Actual: $($matches[1])"
+            if ($_.Node.Version -match $ALPHA_DATE_REGEX) {
+                if ($baselineVersionDate -eq $null) {
+                    Write-Host "Using baseline version date: $($matches[1])"
```

Comment:
Nit: `$matches` is global, and you reference `$matches[1]` numerous times. I'd captured it once in a local and use that throughout.